graphQL/resolvers/mutations.resolvers.js

```diff
@@ -42,48 +40,81 @@ exports.login = async (obj, { email, password }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
-  const { id, title, category } = args;
-  const updatedRoadmap = await db.Roadmaps.update({ title, category },
-    { where: { id } });
-  return updatedRoadmap;
+  const update = await db.Roadmaps.update({ title, category },
+    { where: { id }, returning: true });
+  return update[1][0].dataValues;
 };
 
-exports.deleteRoadmap = async (obj, args, { user }) => {
+exports.deleteRoadmap = async (obj, { id }, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const deletion = await db.Roadmaps.destroy({ where: { id } });
+  if (!deletion) throw new Error('roadmap does not exist');
+  return 'roadmap deleted';
 };
 
-exports.createTopic = async (obj, args, { user }) => {
+exports.createTopic = async (obj, { RoadmapId, title }, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const topic = await db.Topics.create({ title, RoadmapId });
+  return { ...topic.dataValues, checklist: [] };
 };
 
 exports.updateTopic = async (obj, args, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const {
+    id,
+    title,
+    description,
+    resources,
+    completed,
+  } = args;
+
+  const update = await db.Topics.update({
+    title,
+    description,
+    resources,
+    completed,
+  }, { where: { id }, returning: true });
+
+  return update[1][0].dataValues;
 };
 
-exports.deleteTopic = async (obj, args, { user }) => {
+exports.deleteTopic = async (obj, { id }, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const deletion = await db.Topics.destroy({ where: { id } });
+  if (!deletion) throw new Error('topic does not exist');
+  return 'topic deleted';
 };
 
-exports.createChecklistItem = async (obj, args, { user }) => {
+exports.createChecklistItem = async (obj, { TopicId, title }, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const checklistItem = await db.ChecklistItems.create({
+    title,
+    TopicId,
+  });
+  if (!checklistItem) {
+    throw new Error('Internal Server Error: Could not create ChecklistItem');
+  }
+  return { ...checklistItem.dataValues };
 };
 
-exports.updateChecklistItem = async (obj, args, { user }) => {
+exports.updateChecklistItem = async (obj, { id, title, completed }, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const update = await db.ChecklistItems.update({ title, completed },
+    { where: { id }, returning: true });
+  return update[1][0].dataValues;
 };
 
-exports.deleteChecklistItem = async (obj, args, { user }) => {
+exports.deleteChecklistItem = async (obj, { id }, { user }) => {
   if (!user) throw new AuthenticationError('You must be logged in');
+  const deletion = await db.ChecklistItems.destroy({ where: { id } });
+  if (!deletion) throw new Error('checklist item does not exist');
+  return 'item deleted';
```

Comment:
Would be better to return the id of the deleted item, so that you can remove it from the list of checks you have in the client